compaction/compaction_manager.cc

```diff
@@ -376,6 +379,13 @@ future<> compaction_manager::on_compaction_completion(table_state& t, sstables::
     return t.on_compaction_completion(std::move(desc), offstrategy);
 }
 
+bool compaction_task_executor::is_unrepaired_compaction() const {
+    // When a compaciton is set to repair::sstables_repair_state::repaired
+    // explicitly, it means it compacts only repaired sstables. Otherwise the
+    // compaciton might contain unrepaired sstables.
+    return !(_repair_state && (*_repair_state == repair::sstables_repair_state::repaired));
+}
```

Comment:
Alternative: the submitter of a compaction job should specify it.

Reply:
This is not needed. See again: https://github.com/scylladb/scylladb/pull/24291#issuecomment-2998448634

Reply:
We already submit compaction jobs with the tablet_statle_view. The function is_unrepaired_compaction is used only by the wait for the compaction code. I think we can get rid of is_unrepaired_compaction by provider a tablet_statle_view for the wait function too.